xbmc/FileItem.cpp

```diff
@@ -3733,11 +3736,8 @@ bool CFileItem::LoadDetails()
                          this);
     }
     else
-    {
-      db.Close();
       return false;
-    }
-    db.Close();
```

Comment:
is this removal intentional? You don't seem to close the connection anymore

Reply:
Yes, it is. See commit message for explanation.